backend/main.go

```diff
@@ -397,16 +406,50 @@ func (r apiRequest) process(ctx context.Context) (*apiResponse, error) {
 				if policyContext, err := getContext(policy); err != nil {
 					return nil, err
 				} else {
-					response := engine.Validate(ctx, policyContext)
+					response := eng.Validate(ctx, policyContext)
 					apiResponse.Validation = append(apiResponse.Validation, ConvertEngineResponse(response))
 				}
 			}
+			// generate
+			for _, policy := range policies {
+				if policyContext, err := getContext(policy); err != nil {
+					return nil, err
+				} else {
+					response := eng.Generate(ctx, policyContext)
+					if len(response.PolicyResponse.Rules) == 0 {
+						apiResponse.Generation = append(apiResponse.Generation, ConvertEngineResponse(response))
+						continue
+					}
+					gr := toGenerateRequest(policy, resource)
+					ctx, err := engine.NewPolicyContext(jp, resource, operation, &admissionInfo, cfg)
```

Comment:
do we need a new context here ?

Reply:
yes because its not the same interface and I could not reuse the `policyContext`